autoload/crystal/indent.vim

```diff
@@ -0,0 +1,327 @@
+" Variables {{{1
+" =========
+
+" Regex of syntax group names that are strings or characters.
+const crystal#indent#syng_string =
```

Comment:
- `:const` was added at v8.1.1539 and the latest Ubuntu LTS provides 8.0.1453 yet. So I feel it's better to use `var` and `:lockvar` for now
- Please use prefix `g:` for global variables to avoid confusion
- This plugin already uses `g:crystal_*` variables for users configurations. I understood that these global variables in this script are only for implementation, am I correct? Since mixing both `g:crystal_*` and `g:crystal#*` might confuse users, I want to provide `g:crystal_*` variables for user configurations.

Reply:
I see, I didn't realize that `const` was such a recent feature.

As for your third point: I decided to move these variables to `autoload/crystal/indent` and make them global because I needed to use them in both `indent/crystal` and `indent/ecrystal`. Are you saying that we should not make them global? If so, I can move them back to `indent/crystal` and copy the ones I need into `indent/ecrystal`.

Reply:
> Are you saying that we should not make them global?

No, making them global is ok. I just worried about both `g:crystal_*` and `g:crystal#*` variables confusing users. If the `g:crystal#*` variables are not intended to be used by users, current implementation looks good.